geoh5py/ui_json/utils.py

```diff
@@ -94,13 +94,21 @@ def dependency_requires_value(ui_json: dict[str, dict], parameter: str) -> bool:
     :param parameter: Name of parameter to check type.
     """
     dependency = ui_json[parameter]["dependency"]
-    key = "enabled" if ui_json[dependency].get("optional", False) else "value"
+
+    key = (
+        "enabled"
+        if (
+            ui_json[dependency].get("optional", False)
+            or "enabled" in ui_json[dependency]
```

Comment:
I am not sure this is going to work since "enabled" can be set to true, eventhough the groupOptional is set to False.

Reply:
I extended the tests